src/utils/hooks/useTriggerProcessing.tsx

```diff
@@ -1,25 +1,62 @@
 import { useState, useEffect } from 'react'
 import { useGetTriggersQuery } from '../../utils/generated/graphql'
 
-type triggerTypes = 'applicationTrigger' | 'reviewTrigger'
+type TriggerType = 'applicationTrigger' | 'reviewAssignmentTrigger' | 'reviewTrigger'
 
-const useTriggerProcessing = (props: { serialNumber: string; trigger: triggerTypes }) => {
-  const { serialNumber, trigger } = props
+interface TriggerQueryProps {
+  serialNumber?: string
+  reviewAssignmentId?: number
+  reviewId?: number
+  triggerType?: TriggerType
+}
+
+const useTriggerProcessing = ({
+  serialNumber,
+  reviewAssignmentId,
+  reviewId,
+  triggerType,
+}: TriggerQueryProps) => {
   const [isProcessing, setIsProcessing] = useState(true)
   const [triggerError, setTriggerError] = useState(false)
 
+  // Ensure at least one the primary identifiers is provided
+  if (!serialNumber && !reviewAssignmentId && !reviewId) {
+    console.log('INVALID QUERY')
+    setIsProcessing(false)
+    setTriggerError(true)
+  }
+
+  // If triggerType not provided, infer it from the supplied ID
+  const inferredTriggerType = inferTriggerType(triggerType, serialNumber, reviewAssignmentId)
+
   const { data, loading, error } = useGetTriggersQuery({
     variables: {
       serial: serialNumber,
+      reviewAssignmentId,
+      reviewId,
     },
     pollInterval: 500,
     skip: !isProcessing,
     fetchPolicy: 'no-cache',
   })
 
   useEffect((): any => {
-    if (data?.applicationTriggerStates?.nodes[0]) {
-      const triggerRequested = data?.applicationTriggerStates?.nodes[0][trigger]
+    if (data?.applicationTriggerStates?.nodes?.[0]) {
+      let triggerRequested
+      switch (inferredTriggerType) {
+        case 'applicationTrigger':
```

Comment:
Same thing here, if you have a `type` why not use it? It is good for any changes that you might have it would point out for using the wrong string right away. : )

Reply:
And I'm pretty sure there is a more elegant solution for switch in the ES6, have a look on that. And you can probably merge the code for setting the error and processing to `false` in case of bad parameters.

Reply:
> Same thing here, if you have a `type` why not use it? It is good for any changes that you might have it would point out for using the wrong string right away. : )

This is being typed implicitly now, based on last change. `inferredTriggerType` has to be `TriggerType` as that's what its defining function returns. Can confirm that if you try and change one of these "case" strings to something non-conforming, typescript rejects it.

Reply:
> And I'm pretty sure there is a more elegant solution for switch in the ES6, have a look on that.

The trigger field names match the "TriggerType" options, so can just refer to the trigger directly in one line!
```
if (triggers[inferredTriggerType] === null) setIsProcessing(false)
```

Good suggestion, thanks 👍